src/main/java/learnositysdk/request/Telemetry.java

```diff
@@ -0,0 +1,160 @@
+package learnositysdk.request;
+
+import org.apache.commons.io.IOUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.nio.charset.Charset;
+import java.util.Locale;
+import java.util.Properties;
+
+class Telemetry {
+	private static boolean telemetryEnabled = true;
+	private static JSONObject telemetryData = null;
+
+	// Java Runtimes (as expected in lib-metrics)
+	private final static String RT_OPEN_JDK = "openjdk";
+	private final static String RT_JAVA = "java";
+
+	// Operating Systems (as expected in lib-metrics)
+	private final static String OS_DARWIN = "darwin";
+	private final static String OS_LINUX = "linux";
+	private final static String OS_WIN = "win";
+
+	static boolean isEnabled()
+	{
+		return telemetryEnabled;
+	}
+
+	static void disableTelemetry()
+	{
+		Telemetry.telemetryEnabled = false;
+	}
+
+	static void enableTelemetry()
+	{
+		Telemetry.telemetryEnabled = true;
+	}
+
+	/**
+	 * Add the telemetry data to the request packet, under "meta.sdk".
+	 *
+	 * @param requestPacket The request to add the telemetry data to
+	 */
+	static void addToRequest(JSONObject requestPacket)
+	{
+		JSONObject meta;
+
+		if (!Telemetry.telemetryEnabled) {
+			return;
+		}
+
+		try {
+			meta = requestPacket.getJSONObject("meta");
+			meta.put("sdk", Telemetry.getTelemetryData());
+		} catch (JSONException e) {
+			meta = new JSONObject();
+			meta.put("sdk", Telemetry.getTelemetryData());
+			requestPacket.put("meta", meta);
+		}
+	}
+
+	private static JSONObject getTelemetryData()
+	{
+		if (Telemetry.telemetryData != null) {
+			return Telemetry.telemetryData;
+		}
+
+		JSONObject telemetryData = Telemetry.telemetryData = new JSONObject();
+		telemetryData.put("version", Telemetry.getSdkVersion());
+		telemetryData.put("lang", Telemetry.getLanguage());
+		telemetryData.put("lang_version", System.getProperty("java.version"));
+		telemetryData.put("platform", Telemetry.getPlatform());
+		telemetryData.put("platform_version", Telemetry.getPlatformVersion());
+
+		return telemetryData;
+	}
+
+	private static String getSdkVersion()
+	{
+		String version = "unknown";
+		Properties properties = new Properties();
+		InputStream propertiesStream = Telemetry.class.getClassLoader().getResourceAsStream("project.properties");
+
+		if (propertiesStream != null) {
+			try {
+				properties.load(propertiesStream);
+				version = properties.getProperty("version");
+			} catch (Exception e) {
+				// Fail silently, returning default
+			}
+		}
+
+		return version;
+	}
+
+	private static String getLanguage()
+	{
+		String runtime = System.getProperty("java.runtime.name").toLowerCase(Locale.ENGLISH);
+
+		if (runtime.contains(Telemetry.RT_OPEN_JDK)) {
+			return Telemetry.RT_OPEN_JDK;
+		}
+
+		return Telemetry.RT_JAVA;
```

Comment:
Ok, I see how parsing would be an issue, but I think we should have an `unknown` value here, and only return RT_JAVA when positively identified.

Reply:
I'm reluctant to return an "unknown" here because this is how we determine which SDK it's coming from in the first place, unless we were comfortable with determining that by the version number of the SDK alone (which could be unreliable in the future).

A value like `unknown-jvm` might be good, though, because it would be loged as an unparseable value that we can see track down.

Reply:
Yup, I'm happy with that. That said, maybe we'd be better with `lang` to be undconditionally `java`, and lang_version to contain the JDK flavour and version.

Reply:
Sadly we would then lose all of that data without going through and updating lib-metrics everywhere. Thoughts on a `lang_flavor` field? We'll still lose that until we update lib-metrics for parsing, but at least it won't be merged in with other fields then. 😅

Reply:
Let's go for `lang_flavour`, though we won't agree on the spelling there q:

We can update lib-metrics later, but it's the sanest approach 👍